Designing the Perfect Olympic Athlete

Every four years, millions of sports fans from around the world gather to watch their favorite athletes compete in the Olympic Games. We are amazed at the achievements of these athletes and on occasion, we are able to witness true , once in a generation, greatness. Michael Phelps, Larisa Latynina, and Jesse Owens are just a few of the many legends that have been made at the Olympics.

Is it possible, though, to create the perfect Olympian - independent of sport? What would they look like? Where would they be from? Below we will take a look at raw data of listing every Olympian since the founding of the Modern Games in 1896. We will take a look at what makes an Olympian great, and, if it is possible to create the "perfect Olympian."

In [1]:
import pandas as pd
import numpy as np

Our data comes from Kaggle.com's Featured Dataset on Olympians, which we will read as a CSV below.

In [2]:
filepath = '/Users/mal/Downloads/athlete_events.csv'

In [3]:
df = pd.read_csv(filepath)

After reading our CSV into a Pandas Dataframe, we are able to see the columns that make up our data. The columns are:
    
ID:      Athlete ID Number
Name:    Name of the Athlete
Sex:     Gender of the Athlete
Age:     Age in Years
Height:  Height in Centimeters
Weight:  Weight in Kilograms
Team:    Country the Athlete
NOC:     Nationality or Country
Games:   Which Olympic Cycle the Event Took Place In
Year:    Year of the Olympic Games in which the Athlete Competed
Season:  Winter or Summer Olympics
City:    City in which the Olympic Games took Place
Sport:   Sport that the Athlete Competed in for this Event
Event:   Specific Event that the Athlete Competed In
Medal:   Resulting Medal of the Athlete in the listed Event (if any)

As you can see below, our data is a bit mess. We first must remove excess and/or repetitive data to give us a more clear picture of what it is exactly that we are looking at and if our data tells us anything.

In [4]:
df.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [5]:
df.index = df['ID']

In [6]:
del df['ID']
del df['Team']
del df['Games']
del df['City']

As we can see, we have removed excess columns that repeat infomation or provide unneccessary information to give ourselves a cleaner look at the dataframe at hand. Now we must remove athletes for whom we do not have complete data as well as events that did not result in a Medal. It is important to note that many of the Olympians in earlier games do not have a listed height and weight, so this may skew our results towards more recent Olympic Games.

In [7]:
df.head()

Name Sex   Age  Height  Weight  NOC  Year  Season  \
ID                                                                          
1                  A Dijiang   M  24.0   180.0    80.0  CHN  1992  Summer   
2                   A Lamusi   M  23.0   170.0    60.0  CHN  2012  Summer   
3        Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN  1920  Summer   
4       Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN  1900  Summer   
5   Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED  1988  Winter   

            Sport                             Event Medal  
ID                                                         
1      Basketball       Basketball Men's Basketball   NaN  
2            Judo      Judo Men's Extra-Lightweight   NaN  
3        Football           Football Men's Football   NaN  
4      Tug-Of-War       Tug-Of-War Men's Tug-Of-War  Gold  
5   Speed Skating  Speed Skating Women's 500 metres   NaN

In [8]:
df = df.dropna()

We now have a clean dataset showing us the ID number, name, sex, age, height, weight, nationality, year, season, sport, event, and medal of each athlete we will be looking at to create our ultimate olympian.

In [9]:
df.head()

Name Sex   Age  Height  Weight  NOC  Year  Season  \
ID                                                                          
16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  FIN  2014  Winter   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   

         Sport                                   Event   Medal  
ID                                                              
16  Ice Hockey             Ice Hockey Men's Ice Hockey  Bronze  
17  Gymnastics  Gymnastics Men's Individual All-Around  Bronze  
17  Gymnastics        Gymnastics Men's Team All-Around    Gold  
17  Gymnastics            Gymnastics Men's Horse Vault    Gold  
17  Gymnastics        Gymnastics Men's Pommelled Horse    Gold

Now we will assign values to Gold, Silver, and Bronze events. A "Gold" event will be worth 5 points as they contribute the most to the legacy of a great Olympian. A "Silver" event will be worth 3 points, as they contribute substantially less to the legacy of an Olympian as compared to Gold. Finally, a "Bronze" event will be worth 1 point as it contributes substantially less than Silver to the legacy of great Olympians.

In [10]:
df['medal_value'] = pd.np.where(df.Medal.str.contains('Gold'), 5,
                    pd.np.where(df.Medal.str.contains('Silver'), 3,
                    pd.np.where(df.Medal.str.contains('Bronze'), 1, 'medal_value')))

In [11]:
df.head()

Name Sex   Age  Height  Weight  NOC  Year  Season  \
ID                                                                          
16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  FIN  2014  Winter   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   

         Sport                                   Event   Medal medal_value  
ID                                                                          
16  Ice Hockey             Ice Hockey Men's Ice Hockey  Bronze           1  
17  Gymnastics  Gymnastics Men's Individual All-Around  Bronze           1  
17  Gymnastics        Gymnastics Men's Team All-Around    Gold           5  
17  Gymnastics            Gymnastics Men's Horse Vault    Gold           5  
17  Gymnastics        Gymnastics Men's Pommelled Horse    Gold           5

Now we must sort our dataframe into Male and Female Olympians as the Olympics are, in reality, divided into events by gender.

In [12]:
df_male = df[df.Sex == 'M']
df_female = df[df.Sex == 'F']

In [13]:
df_male.head()

Name Sex   Age  Height  Weight  NOC  Year  Season  \
ID                                                                          
16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  FIN  2014  Winter   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   
17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  FIN  1948  Summer   

         Sport                                   Event   Medal medal_value  
ID                                                                          
16  Ice Hockey             Ice Hockey Men's Ice Hockey  Bronze           1  
17  Gymnastics  Gymnastics Men's Individual All-Around  Bronze           1  
17  Gymnastics        Gymnastics Men's Team All-Around    Gold           5  
17  Gymnastics            Gymnastics Men's Horse Vault    Gold           5  
17  Gymnastics        Gymnastics Men's Pommelled Horse    Gold           5

In [14]:
df_female.head()

Name Sex   Age  Height  Weight  NOC  \
ID                                                                            
37                        Ann Kristin Aarnes   F  23.0   182.0    64.0  NOR   
65                         Patimat Abakarova   F  21.0   165.0    49.0  AZE   
67   Mariya Vasilyevna Abakumova (-Tarabina)   F  22.0   179.0    80.0  RUS   
90                 Tamila Rashidovna Abasova   F  21.0   163.0    60.0  RUS   
153                    Monica Cecilia Abbott   F  23.0   191.0    88.0  USA   

     Year  Season      Sport                            Event   Medal  \
ID                                                                      
37   1996  Summer   Football        Football Women's Football  Bronze   
65   2016  Summer  Taekwondo      Taekwondo Women's Flyweight  Bronze   
67   2008  Summer  Athletics  Athletics Women's Javelin Throw  Silver   
90   2004  Summer    Cycling           Cycling Women's Sprint  Silver   
153  2008  Summer   Softball        Softball Women's Softball  Silver   

    medal_value  
ID               
37            1  
65            1  
67            3  
90            3  
153           3

Now that we have our dataframes in male and female groupings, it is time to look at the physical attributes that contribute to the success of Olympic athletes in our data. What is the ideal age of a Male Athlete in the Olympics? A female athlete? What will their height be? Their Weight? In order to find out, we must use our medal values to weigh the outcomes of each event. This will tell us the age, height, and weight with which we will most likely be able to achieve Olympic glory.

In [15]:
def ideal_age(df):
    ideal_age = sum((df.Age.astype(float) * df.medal_value.astype(float))) / sum(df.medal_value.astype(float))
    return ideal_age.round(2)

In [17]:
def ideal_height(df):
    ideal_height = sum((df.Height.astype(float) * df.medal_value.astype(float))) / sum(df.medal_value.astype(float))
    return ideal_height.round(2)

In [18]:
def ideal_weight(df):
    ideal_weight = sum((df.Weight.astype(float) * df.medal_value.astype(float))) / sum(df.medal_value.astype(float))
    return ideal_weight.round(2)

In [19]:
print('Ideal Male Olympian Age:', ideal_age(df_male), 'years old')
print('Ideal Male Olympian Height:', ideal_height(df_male), 'cm')
print('Ideal Male Olympian Weight:', ideal_weight(df_male), 'kg', '\n')
print('Ideal Female Olympian Age:', ideal_age(df_female), 'years old')
print('Ideal Female Olympian Height:', ideal_height(df_female), 'cm')
print('Ideal Female Olympian Weight:', ideal_weight(df_female), 'kg')

Ideal Male Olympian Age: 25.85 years old
Ideal Male Olympian Height: 181.5 pounds
Ideal Male Olympian Weight: 79.4 inches 

Ideal Female Olympian Age: 24.51 years old
Ideal Female Olympian Height: 170.63 pounds
Ideal Female Olympian Weight: 63.32 inches


As we can see above, our perfect Male Olympian is about 26 years old, 182 centimeters tall, and just over 79 kilograms. Our perfect Female Olympian is between 24 and 25 years old, just under 171 centimeters, and just over 63 kilograms. These ages, heights, and weights will make our athlete competitive in a variety of sports and thus will give them the greatest chance of winning, were they to enter every event possible.

Are physical attributes alone enough to create a world renowned Olympic athlete, however? In order to test this, we must look at the countries our athletes come from to determine other factors in play beyond physical makeup.

In [20]:
winningest_countries = pd.value_counts(df.NOC.ravel())

In [24]:
winningest_countries.head()

USA    4383
URS    2246
GER    1612
AUS    1206
RUS    1134
dtype: int64

As we can see, there are clearly countries that dominate the Olympic Games. For example, it is far easier to win an Olympic medal as an athlete from a developed country. The United States and Russia lead the way with 4,383 medals and 3,380 medals (USSR: 2246 + Russia: 1134) respectively. As a result, our athlete must grow up in the 1st world in a develope or increasingly developing country if we want them to have the best chance of success in the Olympic Games. Now that we know where to raise our athlete and what size they should be, what should we name this athlete?

In [72]:
def naming_our_athlete(df):
    ideal_name = sum(df['Name'].str.len() * df.medal_value.astype(float)) / sum(df.medal_value.astype(float))
    return ideal_name.round(2)

In [76]:
print("Our Male Athlete's name should be", naming_our_athlete(df_male), "characters long")
print("Our Female Athlete's name should be", naming_our_athlete(df_female), "characters long")

Our Male Athlete's name should be 19.78 characters long
Our Female Athlete's name should be 23.46 characters long


As we can see, the best male athletes have names with roughly 20 characters and the best femal athletes have names with roughly 23 characters. As a result, we will call our male athlete Marcus Leon Williams who will hail from the United States and our female athlete Nastia Maria Luchenova who will hail from Russia.

Now that we have created our superior athletes, Mr. Marcus Leon Williams and Miss Nastia Maria Luchenova, it is time to put them to the test. See you in Tokyo in 2020!